lib/http_client.dart

```diff
@@ -25,7 +44,13 @@ class HttpClient {
 
   /// Get function invoked on HTTP requests
   Future<void> get(QueryParameters queryParameters, [Duration? timeout]) async {
-    final uri = Uri.https(_baseUri, '/v1/vardata', queryParameters.toJson());
+    final Uri uri;
+    if (_proxyUrl != null) {
+      uri = Uri.parse(_proxyUrl!)
+          .replace(queryParameters: queryParameters.toJson());
+    } else {
+      uri = Uri.https(_baseUri, '/v1/vardata', queryParameters.toJson());
```

Comment:
**consider**: extracting this hardcoded `/v1/vardata` to the static constant instead so that devs can rely on it in their code bases, use it in tests, etc.